.github/workflows/cd.yml

```diff
@@ -9,35 +8,63 @@ on:
   release:
     types:
       - published
-
 concurrency:
-  group: ${{ github.workflow }}-${{ github.ref }}
+  group: "${{ github.workflow }}-${{ github.ref }}"
   cancel-in-progress: true
-
 env:
   FORCE_COLOR: 3
-
 jobs:
   dist:
     name: Distribution build
     runs-on: ubuntu-latest
-
     steps:
       - uses: actions/checkout@v4
         with:
           fetch-depth: 0
 
+      - name: Set up Python
+        uses: actions/setup-python@v5
+        with:
+          python-version: 3.11
+
+      - name: Install dependencies
+        run: |
+          pip install requests==2.31.0 pandas==2.1.1 \
+          google-cloud-bigquery==3.12.0 pyarrow==13.0.0 \
+          db-dtypes==1.1.1 pygithub==2.1.1
```

Comment:
Unfortunately, I do not know where we are invoking/using dev env of pyproject.toml

Reply:
Since these dependencies are only needed to build the wheel and not to use it, you would list them like that:

```diff
[build-system]
-requires = ["scikit-build-core"]
+requires = [
    "scikit-build-core",
    "db-dtypes",
    "google-cloud-bigquery",
    "pandas",
    "pyarrow",
    "pygithub",
    "requests",
]
build-backend = "scikit_build_core.build"
```